### PR TITLE
1711 create official domains for ittms

### PR DESCRIPTION
**Pull request by johnake**
## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [ ] API release notes have been updated if necessary
- [ ] If it adds a significant user-facing change, is it documented in the [CHANGELOG](CHANGELOG.md)?
- [ ] Required environment variables have been updated or [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -53,6 +53,19 @@ ci:
...
+	$(eval include global_config/domain.sh)
+	echo "processed script"
+
+.PHONY: set-azure-template-tag
+set-azure-template-tag:
+	$(eval ARM_TEMPLATE_TAG=1.1.6)
+
+.PHONY: set-azure-resource-group-tags
+set-azure-resource-group-tags: ##Tags that will be added to resource group on its creation in ARM template
+	$(eval RG_TAGS=$(shell echo '{"Product" : "ITT Mentor Services", "Service Offering": "ITT Mentor Services", "Environment" : "$(ENV_TAG)"}' | jq . ))
```
Comment (RMcVelia) [inline Makefile:67]: For product and service offering, can it be changed to "ITT mentor services"

Comment (RMcVelia) [inline Makefile:67]: For product and service offering, can it be changed to "ITT mentor services"

File: Makefile [hunk 2]
```diff
@@ -105,6 +118,23 @@ deploy-arm-resources: arm-deployment ## Validate ARM resource deployment. Usage:
...
+	./bin/terrafile -p terraform/domains/infrastructure/vendor/modules -f terraform/domains/infrastructure/config/zones_Terrafile
+
+	terraform -chdir=terraform/domains/infrastructure init -reconfigure -upgrade
+
+domains-infra-plan: domains-infra-init ## terraform plan for dns core resources
+	terraform -chdir=terraform/domains/infrastructure plan -var-file config/zones.tfvars.json
+
+domains-infra-apply: domains-infra-init ## terraform apply for dns core resources
+	terraform -chdir=terraform/domains/infrastructure apply -var-file config/zones.tfvars.json ${AUTO_APPROVE}
+
```
Comment (RMcVelia) [inline Makefile:137]: you will also need domains-init, domains-plan and domains-apply for the terraform environment domains

File: global_config/domain.sh [hunk 3]
```diff
@@ -0,0 +1,10 @@
+CONFIG=sandbox
+CONFIG_SHORT=sb
+AZURE_SUBSCRIPTION=s189-teacher-services-cloud-production
+RESOURCE_NAME_PREFIX=s189p01
+ENV_TAG=Sandbox
```
Comment (RMcVelia) [inline global_config/domain.sh:5]: if required, then CONFIG, CONFIG_SHORT and ENV_TAG would reference production I would think.

File: terraform/domains/environment_domains/config/sandbox.tfvars.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    "domains": ["apex"],
+    "cached_paths": ["/assets/*"],
+    "environment_short": "sb",
+    "environment_tag": "Sandbox",
+    "origin_hostname": "track-and-pay-sandbox.teacherservices.cloud",
+    "hosted_zone": {
+      "claim-funding-for-mentor-training.education.gov.uk": {
+        "resource_group_name": "s189p01-ittmsdomains-rg",
+        // value to be inputted after CR ?
+        "cnames": {
```
Comment (RMcVelia) [inline terraform/domains/environment_domains/config/sandbox.tfvars.json:11]: cnames shouldn't be required. That was required when using cloudfront, but fd will create the cert validation/auth record

**Review by Nitemaeric — COMMENTED** [review]
Does this link the staging environment to the claim-funding-for-mentor-training.education.gov.uk domain? Or to the staging.claim-funding-for-mentor-training.education.gov.uk domain?

File: terraform/domains/environment_domains/main.tf [hunk 5]
```diff
@@ -0,0 +1,27 @@
+module "domains" {
+  source              = "./vendor/modules/domains//domains/environment_domains"
+  zone                = var.zone
+  front_door_name     = var.front_door_name
+  resource_group_name = var.resource_group_name
+  domains             = var.domains
+  environment         = var.environment_short
+  host_name           = var.origin_hostname
+  null_host_header    = try(var.null_host_header, false)
+  cached_paths        = try(var.cached_paths, [])
```
Comment (RMcVelia) [inline terraform/domains/environment_domains/main.tf:10]: this should refer to the hosted_zone map, and use for_each and each.value

File: terraform/domains/environment_domains/variables.tf [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+}
+
+locals {
+  hostname = "${var.domains[0]}.${var.zone}"
+}
+
+variable "hosted_zone" {
+  type    = map(any)
+  default = {}
+}
```
Comment (RMcVelia) [inline terraform/domains/environment_domains/variables.tf:45]: most of the vars shouldn't be required when using the hosted_zone map

**Review by RMcVelia — APPROVED** [review]

**Pull request merged**